### DISCUSSION TITLE
API and frontend - Interact with and display your database

### DISCUSSION THREAD

**Discussion by V3ntus** [V3ntus/darvester#39] [Announcements]
## Finally!

I too have been anticipating some sort of way to make the gathered data more accessible.

#### Ongoing development has started for two new Darvester projects concerning this issue:
- `darvester-api` - [Github](https://github.com/V3ntus/darvester-api) - Written in Python and FastAPI, this provides a REST API framework that allows quick and easy data retrieval and manipulation over JSON requests
- `darvester-api-frontend` [Github](https://github.com/V3ntus/darvester-api-frontend) - A web frontend written in React.js and MUI that provides an easy to use interface to view (and hopefully manage, in the future) all your database entries in a clean and accessible user interface.

Keep in mind, **these projects are still in alpha phase development**. Contribution is certainly welcome.

**Comment by turnkit** [comment]
For now I just downloaded https://sqlitebrowser.org.  

An interactive way to view things from within the app would be great.  Of course the API is great to connect to other tools.

**Reply by V3ntus** [reply]
I thought about threading the entire harvester process and adding an integrated command line, sort of like bettercap's architecture. The current layout of [harvester.py](https://github.com/V3ntus/darvester/blob/main/src/harvester.py) may or may not allow that easily. I plan to add threading eventually someday, so maybe that will open a door.

For now, I'm just going with API a frontend, which may be bundled into one app soon as well.
